### PR TITLE
Hourly Day of Week Model

### PR DESCRIPTION
**Pull request by toshi09**
Tom,
Please review this model file. I have remove the script file. Will add it later after removing the client name.
Keeping this PR very small.

### CODE REVIEW

**Comment by tplagge** [comment]
I like where you're going with this! A few things:
1. Looks like tests aren't passing on python3 because of things like print() needing parentheses.

2. The other predict() methods take an optional "summed" kwarg and return a data frame with "predicted" and "variance" columns; if you do the same, it'll let the derivatives run without modification.

3. I see you're using tempF as a predictive variable instead of HDD/CDD. Definitely makes things easier, but I'd suggest using HDD and CDD instead with fixed balance points. Otherwise you'd get weird behavior when the training data included both very warm and very cold days; the latter would want to push the tempF coefficient negative, while the former would want to push it positive, and you'd end up getting some funny intermediate value near zero.

4. You don't end up using the WEEK_END and WEEK_DAY constants, right?

**Comment by tplagge** [comment]
Also, you may find this branch interesting:
https://github.com/openeemeter/eemeter/tree/feature/resource-curve

**Comment by toshi09** [comment]
@tplagge  How do I test for python3 locally?, 
1. Will modify the predict method for summed and variance.
2. Removed Weekend and Weekday comp
3. Will do the HDD, CDD comp

**Comment by toshi09** [comment]
@tplagge  please check my previous comment. Also, in the repo mentioned by you, can you point me modules to look into or study?

**Comment by tplagge** [comment]
For Python 3 testing, make sure you have Python 3.6 installed locally, and then do pip3.6 install pytest. Then when you run pytest, as opposed to py.test, it'll run the python3 tests.

In the branch I mentioned, you can take a look at the new models/hourly_load_profile.py file. There's nothing earth-shakingly interesting there, but it's the same thing you're trying to do--get a new model running on hourly data--and I thought it might be a useful example. Take it as you will, no big deal.

**Comment by toshi09** [comment]
@tplagge  I have made changes as per your comment, tested with summed=True and summed=False

File: eemeter/modeling/models/hourly_model.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+        # A Series DS
+        variance_series = self.compute_variance(test_df)
+        if summed:
+            prediction = np.sum(prediction)
+            variance = np.sum(variance_series)
+        return prediction, variance
+
+
+    def forecast(self, df):
```
Comment (philngo) [inline eemeter/modeling/models/hourly_model.py]: What is the functional difference between `predict` and `forecast`? They appear to be quite similar.

**Comment by philngo** [comment]
This is getting close. There are some flake8 issues that need to be fixed before the tests will pass:

```
flake8 runtests: commands[0] | flake8 eemeter --max-line-length=120
eemeter/modeling/models/hourly_model.py:6:1: E302 expected 2 blank lines, found 1
eemeter/modeling/models/hourly_model.py:45:39: E251 unexpected spaces around keyword / parameter equals
eemeter/modeling/models/hourly_model.py:45:41: E251 unexpected spaces around keyword / parameter equals
eemeter/modeling/models/hourly_model.py:111:5: E303 too many blank lines (2)
eemeter/modeling/models/hourly_model.py:122:23: E127 continuation line over-indented for visual indent
eemeter/modeling/models/hourly_model.py:130:23: E127 continuation line over-indented for visual indent
eemeter/modeling/models/hourly_model.py:159:11: E111 indentation is not a multiple of four
eemeter/modeling/models/hourly_model.py:162:9: E303 too many blank lines (2)
eemeter/modeling/models/hourly_model.py:169:9: E303 too many blank lines (2)
eemeter/modeling/models/hourly_model.py:180:1: W391 blank line at end of file
ERROR: InvocationError: '/home/travis/build/openeemeter/eemeter/.tox/flake8/bin/flake8 eemeter --max-line-length=120'
```

**Comment by philngo** [comment]
Tests look great

**Comment by philngo** [comment]
Did you mean to commit changes to `scripts/weather_stations_zipcodes_climate_zones.ipynb`? Those don't seem to fit.

**Comment by toshi09** [comment]
@philngo I will fix the flake8 messages. But can you tell me how to run flake8

**Comment by philngo** [comment]
`flake8 eemeter --max-line-length=120`

**Comment by toshi09** [comment]
@philngo don't mean to commit changes to scripts/weather_stations_zipcodes_climate_zones.ipynb

**Comment by toshi09** [comment]
@philngo  I have addressed the flake8 issues and also took care of notebook script. Please review

**Pull request merged**